src/Auth/AuthContainer.tsx

```diff
@@ -0,0 +1,74 @@
+import React, {useState} from 'react';
+import {Button, TextInput} from 'react-materialize';
+import {auth} from '../Firebase/firebase';
+import firebase from 'firebase/compat/app';
+import {useNavigate} from 'react-router-dom';
+import './style/style.css';
+
+export const AuthContainer = () => {
```

Comment:
There is a special approach in react development, called Container and UI components (or something like that). The point is that you create a component called Container which contains all of the logic (functions, states and so on). Then you create a UI component which only renders UI and you pass props from container to component.

Reply:
it's just fyi